packages/nextjs/src/app/createTenantResolverForClientProvider.ts

```diff
@@ -0,0 +1,17 @@
+import { CustomLoginOptionsType } from '../types';
+
+export const createTenantResolverForClientProvider = ({ subDomainIndex, paramKey }: CustomLoginOptionsType) => {
+  if (subDomainIndex && paramKey) {
+    throw new Error('subDomainIndex and paramKey cannot be used together only one strategy is allowed');
+  }
+  return () => {
+    if (subDomainIndex) {
+      return { tenant: window.location.hostname.split('.')[subDomainIndex] };
```

Comment:
Due to it's for subdomains you have to remove the current main domain from the split
```typescript
window.location.hostname.split('.').slice(0, -2)[subDomainIndex]
``` 

- Wrap this function with try catch
- Return empty object instead of tenant: null if not activated in line 15

Reply:
@frontegg/nextjs: [1]         Type '{ tenant?: undefined; }' is not assignable to type 'ResolvedTenantResult'.

Reply:
what about the second part? 
just to be prepare for next change.
This function should not throw exception no matter what you pass in arguments

Reply:
wrapped the whole function with try catch in case we are on server side and there is no access for the window object